influence_benchmark/environment/assessor_model.py

```diff
@@ -15,61 +15,35 @@ def prepare_messages(self, state) -> List[dict]:
             "environment_system",
         ], "Last message should be from the agent or system"
-        if state.history[-1]["role"] == "environment_system":
-            action = state.history[-2][
-                "content"
-            ]  # We don't want to give feedback on system messages. System messages can only happen after an agent message so this is safe.
-            history = state.history[:-2]
-        else:
-            action = state.history[-1]["content"]
-            history = state.history[:-1]
+        # We don't want to give feedback on the latest system messages. System messages can only happen after an agent message so this is safe.
+        conversation_history = (
+            state.history[:-1] if state.history[-1]["role"] == "environment_system" else state.history
+        )
 
         if perspective == "Third_Person":
             # If feedback is given from the third person perspective, we give the conversation history in one message and ask it to rate it
-            # Importantly, when we can give feedback from the third person pespective,
-            # we still have the choice of whether the feedback is given putting the model into the shoes of the first or third person.
-            messages = []
-            for message in history:
-                if message["role"] == "agent":
-                    messages.append(f"{state.variables['agent_name']}: {message['content']}")
-                elif message["role"] == "environment":
-                    messages.append(f"User: {message['content']}")
-                elif message["role"] == "environment_system":
-                    messages.append(f"System: {message['content']}")
-            conversation_history = "\n\n".join(messages)
-            if len(conversation_history) > 0:
-                prompt = (
-                    f"The conversation history is:\n{conversation_history}"
-                    f"The latest message was from {state.variables['agent_name']} and says:\n\n{state.variables['agent_name']}: {action}\n\n{self.config['prompt_suffix']}"
-                )
-            else:
-                prompt = f"The first message is from {state.variables['agent_name']} and says:\n\n{state.variables['agent_name']}: {action}\n\n{self.config['prompt_suffix']}"
-
-            messages_out = [
-                {"role": "system", "content": self.config["system_prompt"].format(**state.variables).strip()},
-                {"role": "user", "content": prompt},
-            ]
+            task_prompt = self.config["system_prompt"].format(**state.variables).strip()
+
+            context_prompt = "The conversation history is:\n\n"
+            role_to_tag = {
+                "agent": state.variables["agent_name"],
+                "environment": "user",
+                "environment_system": "system",
+            }
+            for message in conversation_history:
+                context_prompt += wrap_between_tags(message["content"], role_to_tag[message["role"]])
```

Comment:
I imagine this will break when we have tool_call and tool_response? should probably add an if to exclude them from conversation history (you removed all the logic which handled this)